rpc/tendermint_handler.go

```diff
@@ -0,0 +1,152 @@
+	ltypes "github.com/loomnetwork/go-loom/types"
+	"github.com/loomnetwork/go-loom/vm"
+
+	"github.com/loomnetwork/loomchain/evm/utils"
+	"github.com/loomnetwork/loomchain/log"
+	"github.com/loomnetwork/loomchain/rpc/eth"
+)
+
+const (
+	ethChainID = "eth"
+	ethTxId = 4
+)
+
+var (
+	blockNumber = big.NewInt(5)
+)
+
+type TendermintPRCFunc struct {
+	eth.HttpRPCFunc
+	name string
+	tm   TendermintRpc
+}
+
+// Tendermint handlers need parameters translated.
+// Only one method supported.
+func NewTendermintRPCFunc(funcName string, tm TendermintRpc) eth.RPCFunc {
+	return &TendermintPRCFunc{
+		name: funcName,
+		tm:   tm,
+	}
+}
+
+func (t *TendermintPRCFunc) UnmarshalParamsAndCall(input eth.JsonRpcRequest, conn *websocket.Conn) (json.RawMessage, *eth.Error) {
+	var tmTx ttypes.Tx
+	if t.name != "eth_sendRawTransaction" {
+		return nil, eth.NewError(eth.EcParseError, "Parse parameters", "unknown method")
+	}
+
+	var jErr *eth.Error
+	ethTx, jErr := t.TranslateSendRawTransactionParams(input)
+	if jErr != nil {
+		return nil, jErr
+	}
+	tmTx, err := ethereumToTendermintTx(ethTx)
+	if err != nil {
+		return nil, eth.NewErrorf(eth.EcServer, "Parse parameters", "convert ethereum tx to tendermint tx, error %v", err)
+	}
+
+	r, err := t.tm.BroadcastTxSync(tmTx)
+	if err != nil {
+		return nil, eth.NewErrorf(eth.EcServer, "Server error", "transaction returned error %v", err)
+	}
+	if r == nil {
+		return nil, eth.NewErrorf(eth.EcServer, "Server error", "transaction returned nil result")
+	}
+
+	var result json.RawMessage
+	result, err = json.Marshal(eth.EncBytes(r.Hash))
+	if err != nil {
+		log.Info("marshal transaction hash %v", err)
+	}
+
+	return result, nil
+}
+
+func (t *TendermintPRCFunc) TranslateSendRawTransactionParams(input eth.JsonRpcRequest) ([]byte, *eth.Error) {
+	paramsBytes := []json.RawMessage{}
+	if len(input.Params) > 0 {
+		if err := json.Unmarshal(input.Params, &paramsBytes); err != nil {
+			return nil, eth.NewError(eth.EcParseError, "Parse params", err.Error())
+		}
+	}
+	var data string
+	if err := json.Unmarshal(paramsBytes[0], &data); err != nil {
+		return nil, eth.NewErrorf(eth.EcParseError, "Parse params", "unmarshal input parameter err %v", err)
+	}
+
+	txBytes, err := eth.DecDataToBytes(eth.Data(data))
+	if err != nil {
+		return nil, eth.NewErrorf(eth.EcParseError, "Parse params", "%v convert input %v to bytes", err, data)
+	}
+
+	return txBytes, nil
+}
+
+func EthToLoomAddress(ethAddr ecommon.Address) loom.Address {
+	return loom.Address{
+		ChainID: ethChainID,
+		Local:   ethAddr.Bytes(),
+	}
+}
+
+func ethereumToTendermintTx(txBytes []byte) (ttypes.Tx, error) {
+	msg := &vm.MessageTx{}
+	msg.Data = txBytes
+
+	var tx types.Transaction
+	if err := rlp.DecodeBytes(txBytes, &tx); err != nil {
+		return nil, err
+	}
+
+	if tx.To() != nil {
+		msg.To = EthToLoomAddress(*tx.To()).MarshalPB()
+	}
+
+	chainConfig := utils.DefaultChainConfig()
+	ethSigner := types.MakeSigner(&chainConfig, blockNumber)
```

Comment:
This picks a signer based on the block number (and you chose 5 for whatever reason), which doesn't make much sense since we're not running an Ethereum node, should figure out which concrete signer implementation we should be using and use that.

Reply:
The block number field is used to see if we are before or after the [`EIP155` fork](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md). 
The block number corresponding to that fork is defined in the ChainConfig, we have set it to [`big.NewInt(0)`](https://github.com/loomnetwork/loomchain/blob/master/evm/evm.go#L263) .
So the blockNumber parameter can be any non-negative integer.
I'll use the value from `chainConfig` to make it more explicit what is going on here.